packages/compiler/react/transform.ts

```diff
@@ -511,20 +549,109 @@ export const transformJSX = (
 
     return dynamics;
   }
+
+  analytics.elements++;
+
   /**
    * Now, it's time to handle the DOM element case.
    */
   const { attributes } = jsx.openingElement;
   for (let i = 0, j = attributes.length; i < j; i++) {
     const attribute = attributes[i]!;
+    analytics.attributes++;
 
     if (t.isJSXSpreadAttribute(attribute)) {
       const spreadPath = jsxPath.get(`openingElement.attributes.${i}.argument`);
-      throw createDeopt(
-        'Spread attributes are not supported.',
-        callSitePath,
+      warn(
+        'Spread attributes are not fully supported.',
         resolvePath(spreadPath),
       );
+      continue;
+    }
+
+    if (
+      t.isJSXIdentifier(attribute.name) &&
+      attribute.name.name === 'style' &&
+      t.isJSXExpressionContainer(attribute.value) &&
+      t.isObjectExpression(attribute.value.expression)
+    ) {
+      const styleObject = attribute.value.expression;
+      const properties = styleObject.properties;
+
+      let hasDynamic = false;
+      for (let i = 0, j = properties.length; i < j; i++) {
+        const property = properties[i]!;
+        if (t.isObjectProperty(property)) {
+          if (property.computed) {
+            // TODO: possibly explore style object extraction in the future.
+            throw createDeopt(
+              'Computed properties are not supported in style objects.',
+              jsxPath,
+            );
+          }
+
+          const value = property.value;
+          if (t.isBooleanLiteral(value) || t.isNullLiteral(value)) {
+            if (t.isNullLiteral(value) || !value.value) {
+              properties.splice(i, 1);
+              i--;
+              j--;
+              // checked={true} -> checked="checked"
+            } else if (t.isIdentifier(property.key)) {
+              property.value = t.stringLiteral(property.key.name);
+            }
+          }
+
+          if (t.isIdentifier(value)) {
+            createDynamic(value, null, null);
+            hasDynamic = true;
+          } else if (t.isLiteral(value) && !t.isTemplateLiteral(value)) {
+            if (t.isStringLiteral(value)) {
+              property.value = value;
+            }
+          } else {
+            const id = createDynamic(null, value as t.Expression, () => {
+              property.value = id;
+            });
+            hasDynamic = true;
+          }
+        } else {
+          hasDynamic = true;
+        }
+      }
+      if (!hasDynamic) {
+        attribute.value = t.stringLiteral(
+          styleObject.properties
+            .map((property) => {
+              if (t.isObjectProperty(property)) {
+                const value = property.value;
+                const key = property.key;
+                if (t.isIdentifier(key) && t.isLiteral(value)) {
+                  let kebabKey = '';
+                  for (let i = 0, j = key.name.length; i < j; ++i) {
+                    const char = key.name.charCodeAt(i);
+                    if (char < 97) {
```

Comment:
Maybe worth it to extract `97` to a top-level const with a proper name, because it's hard to understand what's going on